pkg/daemon/daemon.go

```diff
@@ -372,11 +367,18 @@ func (dn *Daemon) handleErr(err error, key interface{}) {
 }
 
 func (dn *Daemon) updateErrorState(err error) {
-	switch errors.Cause(err) {
+	cause := errors.Cause(err)
+	switch cause {
 	case errUnreconcilable:
-		dn.nodeWriter.SetUnreconcilable(err, dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name)
+		dn.setUnreconcilable(err)
 	default:
-		dn.nodeWriter.SetDegraded(err, dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name)
+		if apierrors.IsConflict(cause) {
```

Comment:
nit: rather than nest an if clause in the switch, remove the switch.
e.g.
```go
	cause := errors.Cause(err)
	if cause == errUnreconcilable {
		return dn.setUnreconcilable(err)
	}
	if apierrors.IsConflict(cause) {
		// skip conflict errors, those are retried and should never result in a permanent
		// failure worth reporting to the node annoation reasons
		glog.Infof("API conflict error: %v, retrying", err)
		return
	}
	dn.setDegraded(err)
}
```